aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,339 @@
+
+For an ``IAsyncActionFilter`` the ``OnActionExecutionAsync`` combines all the possibilities of ``OnActionExecuting`` and ``OnActionExecuted``. A call to ``await next()`` on the ``ActionExecutionDelegate`` will execute any subsequent action filters and the action method, returning an ``ActionExecutedContext``. To short-circuit inside of an ``OnActionExecutionAsync``, assign ``ActionExecutingContext.Result`` to some result instance and do not call the ``ActionExectionDelegate``.
+
+The ``DurationActionFilter`` shown below uses a ``Stopwatch`` to measure the execution time of the action, for display on the page. The ``ViewData`` property of the ``ViewResult`` can be accessed and manipulated, including adding new values.
+
+.. _duration-action-filter:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/DurationActionFilter.cs
+  :language: c#
+  :emphasize-lines: 1,21-26
+  :lines: 8-35
+  :dedent: 4
+
+.. note:: You don't need to use this filter to log action times unless you need to display it on a page. The framework's built-in :doc:`logging </fundamentals/logging>` already logs action timing information.
+
+.. _exception-filters:
+
+Exception Filters
+-----------------
+*Exception Filters* implement either the ``IExceptionFilter`` or ``IAsyncExceptionFilter`` interface.
+
+Exception filters handle unhandled exceptions. They are only called when an exception occurs later in the pipeline. They can provide a single location to implement common error handling policies within an app. The framework provides an abstract `ExceptionFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ExceptionFilterAttribute/index.html>`_ that you should be able to subclass for your needs. Exception filters are good for trapping exceptions that occur within MVC actions, but they're not as flexible as error handling middleware. Prefer middleware for the general case, and use filters only where the middleware doesn't offer the functionality your app requires.
+
+Exception filters do not have two events (for before and after) - they only implement ``OnException`` (or ``OnExceptionAsync``). The ``ExceptionContext`` provided in the ``OnException`` parameter includes the ``Exception`` that occurred. If you set ``context.Exception`` to null, the effect is that you've handled the exception, so the request will proceed as if it hadn't occurred (generally returning a 200 OK status). The following filter logs exceptions and marks them as handled:
```

Comment:
An example we were discussing the other day where we **would** use authorization filters.

Return an "error page" for browser-based endpoints
Return an "error json" for API endpoints

Really, the recommendation is to use exception filters where you need to do error handling *differently* based on which MVC action was chosen.

Default to the middleware if you don't need this.